Added a .clang-format configuration file to enforce the project's con…

dkozinski opened the pull request:
…sistent coding style

Example usage: clang-format -i *.cpp *.h

dkozinski committed:
Added a .clang-format configuration file to enforce the project's consistent coding style

Example usage: clang-format -i *.cpp *.h

dariusz-f commented on .clang-format:
```diff
@@ -0,0 +1,46 @@
+  AfterFunctionDefinitionName: false
+
+SpaceAfterCStyleCast: true
+#SpaceBeforeCStyleCast: true
```
Do not comment line, just switch to false if it required.

dkozinski committed:
Fixes following code review comments

dkozinski committed:
Updated .clang-format configuration

dkozinski committed:
Update .clang-format configuration

- Added a comment recommending clang-format version 19 or higher
- Updated alignment settings for consecutive declarations, assignments, and macros
- Adjusted the column limit from 160 to 120
- Added additional brace wrapping
